src/activities.ts

```diff
@@ -54,25 +55,24 @@ export async function scrapeUrls({ urls: urlsFromFirstAttempt, batchId }: Scrape
   const failedUrls: string[] = []
 
   // TODO: use something like p-props to limit concurrency
-  await Promise.all(
+  await Promise.allSettled(
     urlsToProcess.map(async (url) => {
       try {
         await tryScrape(url)
 
         // Let Temporal know we're still alive for long-running activities (e.g. huge batch size)
-        await Context.current().heartbeat()
+        console.log('scraped ok')
       } catch (error) {
-        console.error(error)
+        console.error('scrape failed', error)
         failedUrls.push(url)
       }
+
+      Context.current().heartbeat({ failedUrlsInLastAttempt: failedUrls })
```

Comment:
What was the motivation behind this?

Reply:
Also if we heartbeat without the failed URLs for the success case, it overwrites the previously reported failed URLs which would be confusing for people looking at the UI.

Reply:
Fair enough.